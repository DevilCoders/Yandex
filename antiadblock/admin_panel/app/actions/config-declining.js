export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

