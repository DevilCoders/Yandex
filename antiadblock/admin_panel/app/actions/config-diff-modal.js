export const OPEN_CONFIG_DIFF_MODAL = 'OPEN_CONFIG_DIFF_MODAL';
export const CLOSE_CONFIG_DIFF_MODAL = 'CLOSE_CONFIG_DIFF_MODAL';

export function openConfigDiffModal(serviceId, id, secondId) {
    return {
        type: OPEN_CONFIG_DIFF_MODAL,
        serviceId,
        id,
        secondId
    };
}

export function closeConfigDiffModal() {
    return {
        type: CLOSE_CONFIG_DIFF_MODAL
    };
}
