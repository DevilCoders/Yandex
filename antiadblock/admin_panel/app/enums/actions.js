export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
