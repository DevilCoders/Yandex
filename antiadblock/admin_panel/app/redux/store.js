import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
