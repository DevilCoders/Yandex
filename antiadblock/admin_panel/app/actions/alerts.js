import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
