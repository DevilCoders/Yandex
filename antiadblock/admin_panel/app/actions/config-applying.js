import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
