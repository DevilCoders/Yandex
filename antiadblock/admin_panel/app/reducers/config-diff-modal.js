import * as configDiffModalActions from 'app/actions/config-diff-modal';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        id: null,
        secondId: null,
        serviceId: null,
        modalVisible: false
    };

    state = state || initialState;

    switch (type) {
        case configDiffModalActions.OPEN_CONFIG_DIFF_MODAL: {
            return {
                id: action.id,
                secondId: action.secondId,
                serviceId: action.serviceId,
                modalVisible: true
            };
        }
        case configDiffModalActions.CLOSE_CONFIG_DIFF_MODAL: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
