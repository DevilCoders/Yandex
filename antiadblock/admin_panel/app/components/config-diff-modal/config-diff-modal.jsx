import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {closeConfigDiffModal} from 'app/actions/config-diff-modal';
import {fetchConfig} from 'app/actions/service';
import {getConfigDiffModal, getService} from 'app/reducers';

import {getConfigs} from 'app/reducers/service';
import DiffModal from 'app/components/diff-modal/diff-modal';

class ConfigDiffModal extends React.Component {
    constructor() {
        super();

        this.onModalClose = this.onModalClose.bind(this);
        this.getLinkDiffConfigToCopyUrl = this.getLinkDiffConfigToCopyUrl.bind(this);
    }

    onModalClose() {
        this.props.closeConfigDiffModal();
    }

    componentDidMount() {
        if (this.props.id && this.props.secondId) {
            this.props.fetchConfig(this.props.serviceId, this.props.id);
            this.props.fetchConfig(this.props.serviceId, this.props.secondId);
        }
    }

    componentWillReceiveProps(nextProps) {
        const idExistsAndDiffers = nextProps.id && nextProps.id !== this.props.id;
        const secondIdExistsAndDiffers = nextProps.secondId && nextProps.secondId !== this.props.secondId;

        // Запрашиваем конфиг только если он есть и если он поменялся
        if (idExistsAndDiffers) {
            this.props.fetchConfig(nextProps.serviceId, nextProps.id);
        }

        if (secondIdExistsAndDiffers) {
            this.props.fetchConfig(nextProps.serviceId, nextProps.secondId);
        }
    }

    getLinkDiffConfigToCopyUrl() {
        const {serviceId, id, secondId} = this.props;

        return `https://${location.host}/service/${serviceId}/configs/diff/${id}/${secondId}`;
    }

    render() {
        const firstConfig = this.props.configsData[this.props.id];
        const secondConfig = this.props.configsData[this.props.secondId];
        const loaded = Boolean(this.props.id && firstConfig && this.props.secondId && secondConfig);

        return (
            <DiffModal
                loaded={loaded}
                firstObj={firstConfig ? firstConfig.data : {}}
                secondObj={secondConfig ? secondConfig.data : {}}
                closeDiffModal={this.onModalClose}
                urlDiff={this.getLinkDiffConfigToCopyUrl()}
                visible={this.props.visible} />
        );
    }
}

ConfigDiffModal.propTypes = {
    id: PropTypes.number,
    secondId: PropTypes.number,
    visible: PropTypes.bool,
    serviceId: PropTypes.string,
    closeConfigDiffModal: PropTypes.func.isRequired,
    fetchConfig: PropTypes.func.isRequired,
    configsData: PropTypes.object
};

export default connect(state => {
    return {
        id: getConfigDiffModal(state).id,
        secondId: getConfigDiffModal(state).secondId,
        visible: getConfigDiffModal(state).modalVisible,
        serviceId: getConfigDiffModal(state).serviceId,
        configsData: getConfigs(getService(state)).byId
    };
}, dispatch => {
    return {
        closeConfigDiffModal: () => {
            dispatch(closeConfigDiffModal());
        },
        fetchConfig: (serviceId, configId) => {
            dispatch(fetchConfig(serviceId, configId));
        }
    };
})(ConfigDiffModal);
