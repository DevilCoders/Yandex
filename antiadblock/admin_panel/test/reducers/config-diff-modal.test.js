import reducer from 'app/reducers/config-diff-modal';
import * as configDiffModalActionFixtures from '../fixtures/actions/config-diff-modal';
import {CONFIG_DIFF_EMPTY, CONFIG_DIFF_OPENED} from '../fixtures/store';

describe('config-diff-modal reducer', () => {
    const initialState = CONFIG_DIFF_EMPTY;
    const openedState = CONFIG_DIFF_OPENED;

    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle OPEN_CONFIG_DIFF', () => {
        expect(
            reducer(initialState, configDiffModalActionFixtures.OPEN_CONFIG_DIFF_MODAL)
        ).toMatchSnapshot();
    });

    it('should handle CLOSE_CONFIG_DIFF', () => {
        expect(
            reducer(openedState, configDiffModalActionFixtures.CLOSE_CONFIG_DIFF_MODAL)
        ).toMatchSnapshot();
    });
});
