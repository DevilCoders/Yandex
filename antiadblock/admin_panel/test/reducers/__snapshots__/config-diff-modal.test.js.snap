// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-diff-modal reducer should handle CLOSE_CONFIG_DIFF 1`] = `
Object {
  "id": null,
  "modalVisible": false,
  "secondId": null,
  "serviceId": null,
}
`;

exports[`config-diff-modal reducer should handle OPEN_CONFIG_DIFF 1`] = `
Object {
  "id": 1,
  "modalVisible": true,
  "secondId": 3,
  "serviceId": "yandex_afisha",
}
`;

exports[`config-diff-modal reducer should return the initial state 1`] = `
Object {
  "id": null,
  "modalVisible": false,
  "secondId": null,
  "serviceId": null,
}
`;
